src/hotp.rs

```diff
@@ -23,20 +23,29 @@ use ring::hmac;
 #[derive(Debug, Eq, PartialEq, Clone)]
 pub struct HOTP {
     /// A secret token for the authentication
-    secret: String,
+    secret: Vec<u8>,
 }
 
 impl HOTP {
     /// Constructs a new `HOTP`
     pub fn new<S: Into<String>>(secret: S) -> HOTP {
-        HOTP { secret: secret.into() }
+        HOTP { secret: secret.into().into_bytes() }
+    }
+
+    pub fn from_base32<S: Into<String>>(secret: S) -> Option<HOTP> {
```

Comment:
Please add doc string for these public methods.

Reply:
Done.